x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/checkpoint/DataFrameTransformsCheckpointService.java

```diff
@@ -96,6 +116,65 @@ public void getCheckpoint(DataFrameTransformConfig transformConfig, long checkpo
 
     }
 
+    public DataFrameTransformCheckpointingInfo getCheckpointStats(DataFrameTransformTask task) {
```

Comment:
It seems to me that this should have an `ActionListener` given that the only place it is used already is already asynchronous.